pkg/server/endpoints/entryfetcher.go

```diff
@@ -49,16 +56,26 @@ func (a *AuthorizedEntryFetcherWithFullCache) FetchAuthorizedEntries(ctx context
 	return a.cache.GetAuthorizedEntries(agentID), nil
 }
 
+func (a *AuthorizedEntryFetcherWithFullCache) FetchAllCachedEntries() ([]*types.Entry, error) {
+	return a.cache.GetAllEntries(), nil
+}
+
 // RunRebuildCacheTask starts a ticker which rebuilds the in-memory entry cache.
 func (a *AuthorizedEntryFetcherWithFullCache) RunRebuildCacheTask(ctx context.Context) error {
 	rebuild := func() {
-		cache, err := a.buildCache(ctx)
-		if err != nil {
+		a.mu.Lock()
```

Comment:
Holding this lock over the duration of the cache update is going to be problematic. It will prevent all authorized entry checks from continuing. The cache itself should gain an internal lock that is held/dropped as needed while updating the cache with each entry change so that the cache can stay responsive to outside callers.

Reply:
Moved the locks to fullcache.go.